src/cueTextTracks.js

```diff
@@ -124,15 +124,15 @@ export function processAdTrack(player, cues, processCue, cancelAds) {
       return;
     }
 
+    // Optional dynamic ad cancellation
+    if (cancelAds) {
+      cancelAds(player, cueData, startTime);
+    }
+
     // Process cue as an ad cue
     processCue(player, cueData, cueId, startTime);
```

Comment:
Well, we are checking for the presence of the adCancel function - the cue in question may not actually be a cancellation cue, in which case it would need to be processed.